src/Common/Models/TaskModel.cs

```diff
@@ -10,20 +10,39 @@ public class TaskModel : BaseModel
         [JsonPropertyName("Name")]
         public override string Name { get; set; }
 
-        [JsonPropertyName("ResourceLock")]
+        [JsonPropertyName("ReplanActionPlannerLocked")]
+        public bool ReplanActionPlannerLocked { get; set; } //Dont change actions in action sequence replan
 
-        public bool ResourceLock { get; set; }
+        [JsonPropertyName("ResourceLock")]
+        public bool ResourceLock { get; set; } //Avoid reusage of resources always.
 
         [JsonPropertyName("TaskPriority")]
         public int TaskPriority { get; set; }
 
         [JsonPropertyName("ActionPlanId")]
-        public Guid ActionPlanId { get; set; }
+        public Guid ActionPlanId { get; set; } //Automatically generated by middleware
+
+        [JsonPropertyName("FullReplan")]
+        public bool FullReplan { get; set; }
+
+        [JsonPropertyName("PartialRePlan")]
+        public bool PartialRePlan { get; set; }
```

Comment:
Maybe we could replace it with the something like `ReplanType` property? We would have only single property instead of 2. 
Exaple values would be:
* None (does not need a replan / have not been replaned)
* FullRelan
* PartialReplan